Code/Framework/AzCore/AzCore/UnitTest/UnitTest.h

```diff
@@ -181,9 +182,37 @@ namespace UnitTest
         }
     };
 
+    class TraceBusHook;
+
+    // Helper class to automatically re-enable the TraceBusHook when going out of scope
+    // This is necessary because the TraceBusHook is part of the Environment which does not reset after each test run
+    // So we need to ensure the bus is always reconnected between tests
+    struct TraceBusHookUniqueDisableToken
+    {
+        TraceBusHookUniqueDisableToken(TraceBusHook* traceBusHook);
+        ~TraceBusHookUniqueDisableToken();
+
+        void ReEnable();
+
+    private:
+        TraceBusHook* m_traceBusHook{};
+    };
+
+    struct TraceBusHookEnvironmentInterface
+    {
+        AZ_RTTI(TraceBusHookEnvironmentInterface, "{DCDC9C73-C8EB-4003-B915-FF8E5D34EF28}");
+        AZ_DISABLE_COPY_MOVE(TraceBusHookEnvironmentInterface);
+
+        TraceBusHookEnvironmentInterface() = default;
+        virtual ~TraceBusHookEnvironmentInterface() = default;
+        
+        virtual AZStd::unique_ptr<TraceBusHookUniqueDisableToken> Disable() = 0;
```

Comment:
Instead of returning a `unique_ptr`, it would be better to return the TraceBusHookUniqueDisableToken by value and just add a default constructor it.
That way it can be used as a member of a class while also allowing a new instance to be copied to that member when the TraceBusHook is desired to be disabled